enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,585 @@
+        }
+      ]
+    }
+  }
+}
+```
+
+#### Notes
+
+ImagePolicy is defined as namespaced CRD. `namespace: openshift-config` is used for cluster policy.
+
+### Risks and Mitigations
+
+Risk: The cluster scoped and namespace policies overwrite
+
+Mitigation: MCO container runtime config controller makes sure namespaced policies for an image cannot override global definitions when merging cluster openshift-config policies and namespaced policies into status field of ImagePolicy instances.
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+MCO container runtime config controller can add unit tests and e2e tests.
+- unit test: verify the policies from ImagePolicy instances merged correctly and the controller writes correct format policy.json
+- e2e test: 
+  - verify the MCO writes configuration to the correct location. 
+  - test demonstrating that an unsigned image in violation of the policy is rejected
+
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade expectations:
+- In order to use ImagePolicy, an existing cluster will have to make an upgrade to the version that the `ImagePolicy` CRD is available.
+- If the existing cluster manages `/etc/containers/policy.json` via the MachineConfig, the configuration will continue to work if no new `ImagePolicy` resources created during the upgrade. If `ImagePolicy` resources created and resulting `/path/to/policies/\<NAMESPACE\>.json` exists, CRI-O will use the config from `/path/to/policies/\<NAMESPACE\>.json`.
+
+Downgrade expectations: 
+- Not applicable. 
+  If `N` does not support ImagePolicy CRD, `N+1`supports IamgePolicy. It is not expected that the ImagePolicy related to the failure `N->N+1`. New resources should not be created during `N->N+1`.
+
+
+### Version Skew Strategy
+
+The MCO does not require skew check.
+The `ImagePolicy` involve coordinating behavior in the CRI-O v1.28. The node team will not expose the `ImagePolicy` CRD if the CRI-O/MCO does not support the namespaced signature to avoid the version skew with CRI-O.
```

Comment:
What does this sentence really mean, what does the feature not being exposed mean?

Reply:
The manifest for this CRD will be removed from openshfit/api if CRI-O/MCO can not support the feature within the target release.

Reply:
Right, I think this could be reworded then. I don't think this is necessarily the purpose of the version skew heading. I think what we are looking for here is context on, what happens during an upgrade, when the CRD is installed and then the CRIO is updated later

Does the policy only take effect when the upgrade is complete?

Reply:
I addressed that the policy will be in effect when both upgrades are completed. In the upgrade process, the initial step reboots the system using a new base image that has an updated CRI-O binary.
 https://github.com/openshift/enhancements/pull/1402/files#diff-fb631ca70e56b7e72dcd1e5e0880976a7a916abf88d8c3c220b7fc2599415f57R558